client/src/components/grants/Show.tsx

```diff
@@ -35,6 +42,34 @@ function Project() {
     }
   }, [dispatch, props.ipfsInitialized, params.id, props.currentGrant]);
 
+  useEffect(() => {
+    async function fetchTimeStamp(projects: ProjectEvent[], projectId: string) {
+      if (global) {
+        const currentProject = projects.find(
+          (project) => project.id === Number(projectId)
+        );
+        if (currentProject) {
+          const blockData = await global.web3Provider?.getBlock(
+            currentProject.block
+          );
```

Comment:
I think we can get the block timestamp in the actions just after retrieving the project so that we have all the logic there and not in the UI,  what do you think?

Reply:
when we get the `ProjectCreated` events, the result already has a `getBlock` function

Reply:
My thought was that by fetching the timestamp on display we won't need to make (project * n) requests before displaying the list of projects

Reply:
we could actually do 1 query filtering by an array of project ids